### PR TITLE
[DDW-1213] Upgrade Electron to 24.2, and Node.js to 18.×

### PR DESCRIPTION
**Pull request by michalrus**
This PR updates Electron and Node.js.

Direct trigger was our need for the `node:crypto` built-in module, which was not available in Node.js 14.×, but it's good to update either way.

Also: all Windows blobs are now cross-built from source (still with Visual Studio).

## Todos

- [x] Linux
- [x] macOS on Intel
- [x] macOS on Apple Silicon
- [x] Windows
- [x] Linux devshell
- [x] macOS devshell

## Screenshots

n/a

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1684859232285079)
- [ ] Please, test hardware wallets
- [ ] And also that opening external links still behaves as expected

**Test Report:** [Link](https://docs.google.com/spreadsheets/d/1l90G2FZoj3VPrQBe4oSjFEBPKhY7UXy_2oHZNT4pR2Q/edit?usp=sharing)

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with typescript types
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by michalrus** [comment]
@gabriela-ponce could you check once more? 🙏  Now it should work (I checked on all platforms)

**Review by gabriela-ponce — APPROVED** [review]
Looks good on macOS (intel and M1) and Windows.
I'd wait for an additional review from @input-output-hk/daedalus-qa (in Linux, preferably) before merging.

**Review by dmitrii-gaico — APPROVED** [review]

**Pull request merged**
